src/common/libidset/idset.h

```diff
@@ -0,0 +1,38 @@
+#ifndef _IDSET_H
+#define _IDSET_H
+
+#include <sys/types.h>
+
+/* An idset is an unordered set of non-negative integers  (0,1,2,3...).
+ */
+
+enum idset_flags {
+    IDSET_FLAG_AUTOGROW = 1, // allow idset to automatically grow beyond 'slots'
+    IDSET_FLAG_BRACKETS = 2, // encode non-singleton idset with brackets
+    IDSET_FLAG_RANGE = 4,    // encode with ranges ("2,3,4,8" -> "2-4,8")
+};
+
+/* Create/destroy an idset.
+ * Set the initial number of slots to 'slots'.
+ * 'flags' may include IDSET_FLAG_AUTOGROW.
+ * Returns idset on success, or NULL on failure with errno set.
+ */
+struct idset *idset_create (size_t slots, int flags);
```

Comment:
Just so that I understand, what are `slots` argument represent, the maximum number of items between delimiters to be supported?

Reply:
Think of the idset as a bitmap.  This is the highest numbered id (plus one) that can be inserted (unless IDSET_FLAG_AUTOGROW is set).  Perhaps that could be clearer.